File: composite/python/Leaf.py
from Component import FileSystemComponent

class File(FileSystemComponent):
    def __init__(self, name, size):
        self.name = name
        self.size = size

    def get_name(self):
        return self.name

    def get_size(self):
        return self.size
    
    def add(self, component):
        raise Exception("Cannot add to a File")
    
    def remove(self, component):
        raise Exception("Cannot remove from a File")

File: composite/php/Leaf.php
<?php

class File implements FileSystemComponent{
    private $name;
    private $size;

    public function __construct($name, $size){
        $this->name = $name;
        $this->size = $size;
    }

    public function getName(){
        return $this->name;
    }

    public function getSize(){
        return $this->size;
    }

    public function add(FileSystemComponent $component){
        throw new Exception("Cannot add to a file");
    }

    public function remove(FileSystemComponent $component){
        throw new Exception("Cannot remove from a file");
    }
}